api/placer.py

```diff
@@ -298,50 +399,53 @@ def finalize(self):
         }
 
         # Get or create a session based on the hierarchy and provided labels.
-        s = {
+        query = {
             'project': bson.ObjectId(self.p_id),
             'label': self.s_label,
             'group': self.g_id
         }
 
-        # self.permissions
-
-        # Add the subject if one was provided
-        new_s = copy.deepcopy(s)
-        subject = self.metadata['session'].get('subject')
-        if subject is not None:
-            new_s['subject'] = subject
-        new_s['modified']    = self.timestamp
-        new_s = util.mongo_dict(new_s)
-
-        # Permissions should always be an exact copy
-        new_s['permissions'] = self.permissions
-
-        session = config.db['session' + 's'].find_one_and_update(s, {
-                '$set': new_s,
-                '$setOnInsert': {
-                    'created': self.timestamp
-                }
+        # Updates if existing
+        updates = {}
+        updates['permissions'] = self.permissions
+        updates['modified']    = self.timestamp
+        updates = util.mongo_dict(updates)
+
+        # Extra properties on insert
+        insert_map = copy.deepcopy(query)
+        insert_map['created'] = self.timestamp
+        insert_map.update(self.metadata['session'])
```

Comment:
This will only add the session properties passed from metadata if the session does not already exist. For example, the subject object will only be set on insert. In the diff it looks like we were previously setting the subject on inserts and updates.